CMakeLists.txt

```
@@ -270,6 +272,24 @@ endif ()
 set(CMAKE_THREAD_PREFER_PTHREAD True)
 find_package(Threads REQUIRED)
 
+if (USE_TRACY)
+    include(FetchContent)
+    add_compile_definitions(TRACY_ENABLE TRACY_ON_DEMAND)
+
+    FetchContent_Declare(
+        tracy
+        GIT_REPOSITORY https://github.com/wolfpld/tracy.git
+        GIT_TAG master
+        GIT_SHALLOW TRUE
+        GIT_PROGRESS TRUE)
+
+    FetchContent_MakeAvailable(tracy)
+
+    if(NOT MSVC)
+        target_compile_options(TracyClient PRIVATE "-Wno-everything")
+    endif()
+endif ()
```

Oh, right, it also needs this for msvc build - should probably work on other builds too
```suggestion
    endif()
    if (NOT DYNAMIC_LINKING)
        set_target_properties(TracyClient PROPERTIES TRACY_STATIC ON)
    endif ()
endif ()
```